### PR TITLE
fix pdf loading preview [sc-13236]

### PR DESCRIPTION
**Pull request by trenchcoder**
[sc-13236]
changed fetching the agreement to use pinata through our backend
added check on file to ensure loading screen waits for agreement to return

### CODE REVIEW

File: pages/agreements/[documentId].tsx [hunk 1]
```diff
@@ -102,7 +102,7 @@ export default function AgreementDetails() {
   // Get the title and the agreement's content from IPFS.
   useEffect(() => {
     if (!documentId) return;
-    const url = `${process.env.NEXT_PUBLIC_IPFS_FILES_URL}/${documentId}`;
+    const url = `${process.env.NEXT_PUBLIC_API_BACKEND_URL}/standard-agreements/ipfs/${documentId}`;
 
     axios.get(url).then((res) => {
```
Comment (CoolHandLuke88) [inline pages/agreements/[documentId].tsx:107]: I know this is not concern, but would extra all calls to the API to a webserviceClient, class.

**Review by CoolHandLuke88 — APPROVED** [review]

File: pages/agreements/[documentId].tsx (hunk 1 above)
Comment (trenchcoder) [reply pages/agreements/[documentId].tsx:107]: absolutely, will address when ticket are created for tech-debt/refactoring

File: pages/agreements/[documentId].tsx [hunk 2]
```diff
@@ -102,7 +102,7 @@ export default function AgreementDetails() {
   // Get the title and the agreement's content from IPFS.
   useEffect(() => {
     if (!documentId) return;
-    const url = `${process.env.NEXT_PUBLIC_IPFS_FILES_URL}/${documentId}`;
+    const url = `${process.env.NEXT_PUBLIC_API_BACKEND_URL}/standard-agreements/ipfs/${documentId}`;
```
Comment (gkmblade) [inline pages/agreements/[documentId].tsx:105]: @trenchcoder We should stop using `process` and use the `env-var` library to clean things up on the next refactor.

**Review by gkmblade — APPROVED** [review]
Approve with a minor comment.

**Pull request merged**